test/core/codec/cbor/cbor_test.cpp

```diff
@@ -0,0 +1,462 @@
+  CborEncodeStream s;
+  auto map1 = s.map();
+  map1["a"] << 1 << 2;
+  EXPECT_OUTCOME_RAISE(CborEncodeError::EXPECTED_MAP_VALUE_SINGLE, CborEncodeStream() << map1);
+  auto map2 = s.map();
+  map2["a"];
+  EXPECT_OUTCOME_RAISE(CborEncodeError::EXPECTED_MAP_VALUE_SINGLE, CborEncodeStream() << map2);
+}
+
+/**
+ * @given Integer and bool CBOR
+ * @when Decode integer and bool
+ * @then Decoded as expected
+ */
+TEST(CborDecoder, Integral) {
+  expectDecodeOne("00"_unhex, 0ull);
+  expectDecodeOne("00"_unhex, 0ll);
+  expectDecodeOne("01"_unhex, 1);
+  expectDecodeOne("17"_unhex, 23);
+  expectDecodeOne("1818"_unhex, 24);
+  expectDecodeOne("20"_unhex, -1);
+  expectDecodeOne("F4"_unhex, false);
+  expectDecodeOne("F5"_unhex, true);
+}
+
+/**
+ * @given CID CBOR
+ * @when Decode CID
+ * @then Decoded as expected
+ */
+TEST(CborDecoder, Cid) {
+  EXPECT_OUTCOME_TRUE_2(expected, ContentIdentifierCodec::decode(kCidRaw));
+  // libp2p::multi::ContentIdentifier is not default constructible and must be initialized somehow
+  EXPECT_OUTCOME_TRUE_2(actual, ContentIdentifierCodec::decode(kCidEmptyRaw));
+  EXPECT_NE(actual, expected);
+  CborDecodeStream(kCidCbor) >> actual;
+  EXPECT_EQ(actual, expected);
+}
+
+/**
+ * @given CID CBOR
+ * @when Skip CID
+ * @then Skipped as expected
+ */
+TEST(CborDecoder, CidNext) {
+  auto bytes = kCidCbor;
+  bytes.push_back(0x01);
+  CborDecodeStream s(bytes);
+  s.next();
+  int i;
+  s >> i;
+}
+
+/**
+ * @given Sequence CBOR
+ * @when Decode sequence
+ * @then Decoded as expected
+ */
+TEST(CborDecoder, Flat) {
+  CborDecodeStream s("0504"_unhex);
+  int a, b;
+  s >> a >> b;
+  EXPECT_EQ(a, 5);
+  EXPECT_EQ(b, 4);
+}
+
+/**
+ * @given List CBOR
+ * @when Decode list container
+ * @then Decoded as expected
+ */
+TEST(CborDecoder, List) {
+  CborDecodeStream s1("82050403"_unhex);
+  auto s2 = s1.list();
+  int a, b;
+  s2 >> a >> b;
+  EXPECT_EQ(a, 5);
+  EXPECT_EQ(b, 4);
+  int c;
+  s1 >> c;
+  EXPECT_EQ(c, 3);
+}
+
+/**
+ * @given String CBOR
+ * @when Decode string
+ * @then Decoded as expected
+ */
+TEST(CborDecoder, String) {
+  std::string s;
+  CborDecodeStream("63666F6F"_unhex) >> s;
+  EXPECT_EQ(s, "foo");
+}
+
+/**
+ * @given Map CBOR
+ * @when Decode map container
+ * @then Decoded as expected
+ */
+TEST(CborDecoder, Map) {
+  auto m = CborDecodeStream("A261610261628101"_unhex).map();
+  int a, b;
+  m.at("a") >> a;
+  m.at("b").list() >> b;
+}
+
+/**
+ * @given Invalid CBOR
+ * @when Init decoder
+ * @then Error
+ */
+TEST(CborDecoder, InitErrors) {
+  EXPECT_OUTCOME_RAISE(CborDecodeError::INVALID_CBOR, CborDecodeStream("FF"_unhex));
+  EXPECT_OUTCOME_RAISE(CborDecodeError::INVALID_CBOR, CborDecodeStream("18"_unhex));
+}
+
+/**
+ * @given Invalid CBOR or wrong type
+ * @when Decode integer and bool
+ * @then Error
+ */
+TEST(CborDecoder, IntErrors) {
+  bool b;
+  uint8_t u8;
+  int8_t i8;
+  EXPECT_OUTCOME_RAISE(CborDecodeError::WRONG_TYPE, CborDecodeStream("01"_unhex) >> b);
```

Comment:
is this file clang-formatted? Usually our clang format does not allow such long rows